src/ScopeProvider/ScopeProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import { Provider, useStore } from 'jotai/react';
+import { type ReactNode, createContext, useContext, useState } from 'react';
+import { createScope, type Scope } from './scope';
+import { AnyAtom, Store } from './types';
+
+const ScopeContext = createContext<{
+  scope: Scope | undefined;
+  baseStore: Store | undefined;
+}>({
+  scope: undefined,
+  baseStore: undefined,
+});
+
+const patchedStoreSymbol = Symbol();
+
+export function ScopeProvider({
+  atoms,
+  children,
+}: {
+  atoms: Iterable<AnyAtom>;
+  children: ReactNode;
+}) {
+  const parentStore = useStore();
+  let { scope: parentScope, baseStore = parentStore } =
+    useContext(ScopeContext);
+  if (!(patchedStoreSymbol in parentStore)) {
+    parentScope = undefined;
+    baseStore = parentStore;
+  }
+
+  /**
+   * atomSet is used to detect if the atoms prop has changed.
+   */
+  const atomSet = new Set(atoms);
+
+  function initialize() {
+    const scope = createScope(atoms, parentScope);
+    const patchedStore: Store & { [patchedStoreSymbol]: true } = {
+      // TODO: update this patch to support devtools
+      ...baseStore,
```

Comment:
we want to use the baseStore (the nearest non-patched store) instead of the parent patched store.